cloud/util/util.go

```diff
@@ -155,13 +155,22 @@ func InitClientsAndRegion(ctx context.Context, client client.Client, defaultRegi
 			return nil, "", scope.OCIClients{}, err
 		}
 		clusterRegion = region
+	} else if clusterAccessor.GetClientHostUrls() != nil {
+		// IdentityRef provider will be created with client host url overrides
+		// but if no identityRef we will want to create a new client provider with the overrides
+		clientProvider, err = scope.NewClientProvider(defaultClientProvider.GetAuthProvider(), clusterAccessor.GetClientHostUrls())
```

Comment:
can we refactor to get the conf based on an if condition , and then fo client provider ? Can you see if the code will be cleaner that way?

Reply:
resolving convo for know as we will have a think on how to refactor this.